tests/functional/conftest.py

```diff
@@ -0,0 +1,107 @@
+import subprocess
+import pytest
+import socket
+import os
+import tempfile
+import shutil
+CONTAINER_NAME="openvino_backend_pytest"
+
+
+@pytest.fixture(scope="class")
+def triton_server(model_repository):
+    port = 0
+    try:
+        sock = socket.socket()
+        sock.bind(('', 0))
+        port = sock.getsockname()[1]
+        sock.close()
+    except socket.error as e:
+        if e.errno != errno.EADDRINUSE:
+            raise Exception("Not expected exception found in port manager {}: {}".format(self.name, e))
+    image_name = os.environ.get("TIS_IMAGE_NAME")
+    image_name = image_name if image_name != None else "tritonserver:latest"
+    subprocess.run(f"docker run -p {port}:8001 -d -v {model_repository}:/model_repository --name={CONTAINER_NAME} --device /dev/dri --group-add=$(stat -c \"%g\" /dev/dri/render* ) {image_name} bin/tritonserver --model-repository /model_repository", capture_output=True, shell=True)
+    subprocess.run(["sleep", "3"])
+    subprocess.run(["docker", "logs", CONTAINER_NAME])
+    yield port
+    subprocess.run(["docker", "stop", CONTAINER_NAME])
+    subprocess.run(["docker", "remove", CONTAINER_NAME])
+
+def copy_config(repo, name, gpu=False):
+    gpu_suffix = "_gpu" if gpu else ""
+    shutil.copy(f"configs/{name}{gpu_suffix}.pbtxt", f"{repo}/{name}/config.pbtxt")
+
+def ir_model(repo):
+    os.makedirs(f"{repo}/ir/1")
+    subprocess.run(f"curl https://storage.openvinotoolkit.org/repositories/open_model_zoo/2022.1/models_bin/2/resnet50-binary-0001/FP32-INT1/resnet50-binary-0001.bin https://storage.openvinotoolkit.org/repositories/open_model_zoo/2022.1/models_bin/2/resnet50-binary-0001/FP32-INT1/resnet50-binary-0001.xml -o {repo}/ir/1/model.bin -o {repo}/ir/1/model.xml", shell=True)
```

Comment:
To make the code clearer, you can create the 'cmd' variable first and use it later